addons/survey/static/src/js/survey_form.js

```diff
@@ -161,6 +172,75 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend({
         this._submitForm({'previousPageId': event.data.previousPageId});
     },
 
+    /**
+     * We listen to 'next_question' and 'end_session' events to load the next
+     * page of the survey automatically, based on the host pacing.
+     *
+     * If the trigger is 'next_question', we handle some extra computation to find
+     * a suitable "fadeInOutDelay" based on the delay between the time of the question
+     * change by the host and the time of reception of the event.
+     * This will allow us to account for a little bit of server lag (up to 2 seconds)
+     * while giving everyone a fair experience on the quiz.
+     *
+     * e.g 1:
+     * - The host switches the question
+     * - We receive the event 500 ms later due to server lag
+     * - -> The fadeInOutDelay will be 750 ms (500ms delay + 750ms * 2 fade in fade out)
+     *
+     * e.g 2:
+     * - The host switches the question
+     * - We receive the event 1500 ms later due to bigger server lag
+     * - -> The fadeInOutDelay will be 250ms (1500ms delay + 250ms * 2 fade in fade out)
+     *
+     * @private
+     * @param {Array[]} notifications structured as specified by the bus feature
+     */
+    _onNotification: function (notifications) {
+        var nextPageEvent = false;
+        if (notifications && notifications.length !== 0) {
+            notifications.forEach(function (notification) {
+                if (notification.length >= 2) {
+                    var event = notification[1];
+                    if (event.type === 'next_question' ||
+                        event.type === 'end_session') {
+                        nextPageEvent = event;
+                    }
+                }
+            });
+        }
+
+        if (this.options.isStartScreen && nextPageEvent.type === 'end_session') {
+            // can happen when triggering the same survey session multiple times
+            // we received an "old" end_session event that needs to be ignored
+            return;
+        }
+
+        if (nextPageEvent) {
+            if (nextPageEvent.type === 'next_question') {
+                var serverDelayMS = moment.utc().valueOf() - moment.unix(nextPageEvent.question_start).utc().valueOf();
+                if (serverDelayMS < 0) {
+                    serverDelayMS = 0;
+                } else if (serverDelayMS > 2000) {
+                    serverDelayMS = 2000;
+                }
+                this.fadeInOutDelay = (2000 - serverDelayMS) / 2;
+            } else {
+                this.fadeInOutDelay = 400;
+            }
+
+            this.preventEnterSubmit = false;
+            this.readonly = false;
+            this._nextScreen(
+                this._rpc({
+                    route: '/survey/next_question/' + this.options.surveyToken + '/' + this.options.answerToken,
```

Comment:
Sometimes you use sprintf, sometimes manual build... somehow you ahve to find your inner balance.

Reply:
Trying to get used to template strings now.
I find them much more readable.